front_end/src/utils/permissions.ts

```diff
@@ -0,0 +1,34 @@
+import { CommentType } from "@/types/comment";
+import { ProjectPermissions } from "@/types/post";
+import { CurrentUser } from "@/types/users";
+
+/**
+ * a comment is editable if:
+ * 1. you are logged in, and:
+ *   2. you are the comment author, or
+ *   3. the comment is on a post in a project you administer or curate
+ */
+export function isEditable(
+  comment: CommentType,
+  user: CurrentUser | null,
+  projectPermissions?: ProjectPermissions
+): boolean {
+  if (!user?.id) {
+    return false;
+  }
+
+  if (comment.author.id === user.id) {
+    return true;
+  }
+
+  return isProjectAdmin(projectPermissions);
```

Comment:
Our backend currently allows comment editing for either the comment author or users with User.is_staff status:
https://github.com/Metaculus/metaculus/blob/2aefa215a78f84255929e6ec0f916a7c8081363e/comments/views.py#L148
However, User.is_staff is a global flag unrelated to Project-specific permissions

To avoid confusion, we would like to restrict comment editing exclusively to the original author. Could you please update your logic accordingly and remove [request.user.is_staff](https://github.com/Metaculus/metaculus/blob/2aefa215a78f84255929e6ec0f916a7c8081363e/comments/views.py#L148) checks in the backend?

Reply:
well, that simplifies things! sure.

Reply:
done!